### DISCUSSION TITLE
I am stuck at running function testConvertSvgToUri()

### DISCUSSION THREAD

**Discussion by liumOazed** [Cyfrin/foundry-full-course-cu#2898] [Q&A] [answered]
Hi guys! I am trying to run the `forge test --mt testConvertSvgToUri -vvvv` but getting the following error:

zed@DESKTOP-1TN6EKN:~/foundry-nft$ forge test --mt testConvertSvgToUri -vvvv
[⠊] Compiling...
[⠒] Compiling 1 files with Solc 0.8.27
[⠑] Solc 0.8.27 finished in 1.46s
Compiler run successful!

Ran 1 test for test/DeployMoodNftTest.t.sol:DeployMoodNftTest
[FAIL: panic: assertion failed (0x01)] testConvertSvgToUri() (gas: 31269)
Logs:
  data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCI+CiAgICA8dGV4dCB4PSIwIiB5PSIxNSIgZmlsbD0iYmxhY2siPkhpISBZb3VyIGJyb3dzZXIgZGVjb2RlZCB0aGlzPC90ZXh0Pgo8L3N2Zz4=

Traces:
  [324704] DeployMoodNftTest::setUp()
    ├─ [287030] → new DeployMoodNft@0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f
    │   └─ ← [Return] 1323 bytes of code
    └─ ← [Stop] 

  [31269] DeployMoodNftTest::testConvertSvgToUri()
    ├─ [17846] DeployMoodNft::svgToImageURI("<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"500\" height=\"500\"><text x=\"0\" y=\"15\" fill=\"black\">Hi! Your browser decoded this</text></svg>") [staticcall]
    │   └─ ← [Return] "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCI+PHRleHQgeD0iMCIgeT0iMTUiIGZpbGw9ImJsYWNrIj5IaSEgWW91ciBicm93c2VyIGRlY29kZWQgdGhpczwvdGV4dD48L3N2Zz4="
    ├─ emit log_string(val: "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCI+CiAgICA8dGV4dCB4PSIwIiB5PSIxNSIgZmlsbD0iYmxhY2siPkhpISBZb3VyIGJyb3dzZXIgZGVjb2RlZCB0aGlzPC90ZXh0Pgo8L3N2Zz4=")
    └─ ← [Revert] panic: assertion failed (0x01)

Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 735.40µs (319.20µs CPU time)

Ran 1 test suite in 1.35s (735.40µs CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/DeployMoodNftTest.t.sol:DeployMoodNftTest
[FAIL: panic: assertion failed (0x01)] testConvertSvgToUri() (gas: 31269)

Encountered a total of 1 failing tests, 0 tests succeeded

Here is my `testConvertSvgToUri` function:

```
 function testConvertSvgToUri() public {
        string
            memory expectedUri = "data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCI+CiAgICA8dGV4dCB4PSIwIiB5PSIxNSIgZmlsbD0iYmxhY2siPkhpISBZb3VyIGJyb3dzZXIgZGVjb2RlZCB0aGlzPC90ZXh0Pgo8L3N2Zz4=";

        string
            memory svg = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="500" height="500"><text x="0" y="15" fill="black">Hi! Your browser decoded this</text></svg>';

        string memory actualUri = deployer.svgToImageURI(svg);

        // Log actual URI for debugging
        emit log_string(expectedUri);
        assert(
            keccak256(abi.encodePacked(actualUri)) ==
                keccak256(abi.encodePacked(expectedUri))
        );
    }
```

Below is my `DeployMoodNft.s.sol` code:

```
//SPDX-License Identifier: MIT
pragma solidity ^0.8.18;
import {Script} from "forge-std/Script.sol";
import {MoodNft} from "src/MoodNft.sol";
import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";

contract DeployMoodNft is Script {
    function run() external returns (MoodNft) {}

    function svgToImageURI(
        string memory svg
    ) public pure returns (string memory) {
        // example:
        // <svg viewBox="0 0 200 200" width="400"....>
        // data:image/svg+xml;base64,PHN2ZyB2a...
        string memory baseURL = "data:image/svg+xml;base64,";
        string memory svgBase64Encoded = Base64.encode(
            bytes(string(abi.encodePacked(svg)))
        );
        return string(abi.encodePacked(baseURL, svgBase64Encoded));
    }
}
```

**Comment by EngrPips** [comment, marked as answer]
Hello @liumOazed, Set your `expectedUri` to the below and retry your test. let's see how that plays out.

```js
"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCI+PHRleHQgeD0iMCIgeT0iMTUiIGZpbGw9ImJsYWNrIj5IaSEgWW91ciBicm93c2VyIGRlY29kZWQgdGhpczwvdGV4dD48L3N2Zz4="
```

**Reply by liumOazed** [reply]
Hey @EngrPips it worked like a charm! Thanks man!

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.

**Reply by BhamlaPrince** [reply]
then why mine that i had created didn't worked

**Reply by EngrPips** [reply]
You must have a slight difference in the data encoded. You can take your time to figure it out if you are curious.
